packages/aragon-wrapper/src/index.js

```diff
@@ -438,6 +464,36 @@ export default class Aragon {
     this.forwarders.connect()
   }
 
+  /**
+   * Initialise app identifier observable.
+   *
+   * @return {void}
+   */
+  initAppIdentifiers () {
+    this.identifiers = new BehaviorSubject({}).pipe(
+      scan(
+        (identifiers, { address, identifier }) =>
+          Object.assign(identifiers, { [address]: identifier })
+      ),
+      publishReplay(1)
+    )
+    this.identifiers.connect()
+  }
+
+  /**
+   * Set the identifier of an app.
+   *
+   * @param {string} address The proxy address of the app
+   * @param {string} identifier The identifier of the app
+   * @return {void}
+   */
+  setAppIdentifier (address, identifier) {
+    this.identifiers.next({
+      address,
+      identifier
+    })
+  }
+
```

Comment:
Note that these changes to the `identifiers` means that this is a breaking change, as the identifier and app lists are not merged by default anymore.

I hadn't realized that the previous logic could create infinite loops until I actually encountered them:

1. Init apps, get apps observable
1. Client starts apps
1. Apps emit on the identifier observable
1. Due to the `combineLatest()` logic, another apps emission would occur
1. If the client then restarts the app due to the apps emission (e.g. in the case of an upgrade), the app could re-emit its identifier
1. Repeat from 4

Hence, I've taken the easy way out by splitting the two concepts from each other and pushing the work to clients.

Reply:
Since it's a breaking change, I also took the liberty of renaming `identifier` to `appIdentifier` to make it more clear at a glance what this observable is for.